### ISSUE TITLE
Executor flags cannot be changed ever in a running cluster

### ISSUE THREAD

**Issue by sttts** [mesosphere/kubernetes-mesos#506]
Changed configuration leads to a different "km minion" command line. Mesos does a deep comparison of the ExecutorInfo and refused to launch a task with TASK_ERROR if the ExecutorInfo changed. Hence, we cannot deploy to a node with a running executor.

A solution would be to use another medium for "km minion" parameters, e.g. via the fetch or via a HTTP interfaces in the scheduler.

**Comment by jdef** [comment]
... and hope that the URI for the scheduler doesn't change, otherwise
you'll have the same problem.

**Comment by sttts** [comment]
mesos-dns to the rescue.

**Comment by jdef** [comment]
Yes, that and ip-per-container. We currently use static scheduler ports but
eventually we'll need to use dynamic ones. Would be great if we could use
SRV records to access URI endpoints for downloading things via the fetcher.
Of course, mesos-dns would need to have updated records relatively quickly
(and there is some lag here already with it's current master-polling
implementation) to avoid a storm of initial task launch failures.

I think @tsenart wrote a proof-of-concept HTTP server that helps out with
SRV-based lookups. Probably compat w/ the current fetcher. Wonder if it
makes sense to integrate with DCOS?

**Comment by tsenart** [comment]
https://github.com/mesosphere/redir
